docs/5-design-patterns/hometask/shipment.ts

```diff
@@ -0,0 +1,78 @@
+import { ShipmentInfo } from './types';
+
+let shipmentId = 1;
+
+export class Shipment {
+    private static rate = 0.39;
+
+    private shipmentId: number;
+    private weight: number;
+    private cost: number;
+    private fromAddress: string;
+    private fromZipCode: string;
+    private toAddress: string;
+    private toZipCode: string;
+
+    constructor({
```

Comment:
We have a requirement in the first task that the `Shipment` should have only a single instance. (be a `singleton`).